networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/Eth2Peer.java

```diff
@@ -171,9 +180,16 @@ void markChainValidated() {
   }
 
   public SafeFuture<UnsignedLong> sendPing() {
-    return requestSingleItem(rpcMethods.ping(), metadataMessagesFactory.createPingMessage())
-        .thenApply(PingMessage::getSeqNumber)
-        .thenPeek(this::updateMetadataSeqNumber);
+    if (outstandingPings.getAndIncrement() >= outstandingPingsThreshold) {
```

Comment:
I'd probably move this logic into `Eth2PeerManager` - the other peer validation / management logic is in the manager.

Reply:
Yep, that makes sense 👍